src/test/java/graphql/annotations/GraphQLInputTest.java

```diff
@@ -191,6 +250,19 @@ public void query() {
         assertEquals(((Map<String, Map<String, String>>) result.getData()).get("object").get("value"), "testa");
     }
 
+    @Test
+    public void queryWithUndefinableParameters() {
+        GraphQLSchema schema = newAnnotationsSchema().query(QueryUndefinedParameter.class).build();
+
+        GraphQL graphQL = GraphQL.newGraphQL(schema).build();
+        ExecutionResult result = graphQL.execute("{ something(code: {firstField:\"a\",secondField:\"b\"}) otherthing(code: {secondField:\"c\"}) listthings somethingWithOneField(code: {}) }", new QueryUndefinedParameter());
+        assertTrue(result.getErrors().isEmpty());
+        assertEquals(((Map<String, String>) result.getData()).get("something"), "ab");
+        assertEquals(((Map<String, String>) result.getData()).get("otherthing"), "c");
+        assertEquals(((Map<String, String>) result.getData()).get("listthings"), "was null");
+        assertEquals(((Map<String, String>) result.getData()).get("somethingWithOneField"), "was undefined");
+    }
+
```

Comment:
Since the goal of this story (from my understanding) is to make the distinction between {} and {value: null} , we should at least have a test that shows it. The use cases described here already work without the proposed changes.